subprojects/problems/src/main/java/org/gradle/api/problems/internal/DefaultProblem.java

```diff
@@ -0,0 +1,141 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.problems.internal;
+
+import org.gradle.api.NonNullApi;
+import org.gradle.api.problems.interfaces.DocLink;
+import org.gradle.api.problems.interfaces.Problem;
+import org.gradle.api.problems.interfaces.ProblemGroup;
+import org.gradle.api.problems.interfaces.ProblemLocation;
+import org.gradle.api.problems.interfaces.Severity;
+
+import javax.annotation.Nullable;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+@NonNullApi
+public class DefaultProblem implements Problem {
+
+    private ProblemGroup problemGroup;
+    private String message;
+    private Severity severity;
+    private ProblemLocation where;
+    private DocLink documentationLink;
+    private String description;
+    private List<String> solutions;
+    private Throwable cause;
+    private String problemType;
+    private Map<String, String> additionalMetadata;
+
+    public DefaultProblem(ProblemGroup problemGroup, String message, Severity severity, @Nullable ProblemLocation location, @Nullable DocLink documentationUrl,
+                          @Nullable String description, @Nullable List<String> solutions, @Nullable Throwable cause, String problemType, Map<String, String> additionalMetadata) {
+        this.problemGroup = problemGroup;
+        this.message = message;
+        this.severity = severity;
+        this.where = location;
+        this.documentationLink = documentationUrl;
+        this.description = description;
+        this.solutions = solutions == null ? Collections.<String>emptyList() : solutions;
+        this.cause = cause;
+        this.problemType = problemType;
+        this.additionalMetadata = additionalMetadata;
+    }
+
+    public DefaultProblem(){
+    }
+    @Override
+    public ProblemGroup getProblemGroup() {
+        return problemGroup;
+    }
+
+    @Override
+    public String getMessage() {
+        return message;
+    }
+
+    @Override
+    public Severity getSeverity() {
+        return severity;
+    }
+
+    @Override
+    public ProblemLocation getWhere() {
+        return where;
+    }
+
+    @Nullable
+    @Override
+    public DocLink getDocumentationLink() {
+        return documentationLink;
+    }
+
+    @Override
+    public String getDescription() {
+        return description;
+    }
+
+    @Override
+    public List<String> getSolutions() {
+        return solutions;
+    }
+
+    @Override
+    public Throwable getCause() {
+        return cause;
+    }
+
+    @Override
+    public String getProblemType() {
+        return problemType;
+    }
+
+    @Override
+    public Map<String, String> getAdditionalMetadata() {
+        return additionalMetadata;
+    }
+
+    private static boolean equals(@Nullable  Object a, @Nullable Object b) {
+        return (a == b) || (a != null && a.equals(b));
+    }
```

Comment:
🤔 Replace this with [`Objects.equals`](https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#equals-java.lang.Object-java.lang.Object-)

Reply:
Not available in java 6 :)